### PR TITLE
Skip ONNX optimization when it fails

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
During ONNX export, model optimization is performed. Currently, if optimization fails, the export also fails

This PR allows export to succeed even when the optimization fails. This is useful to unblock users and allow them to inspect the ONNX model actquired through the export process

A warning with a link to report the bug is printed to let user know about the issue

image

### CODE REVIEW

**Review by titaiwangms — APPROVED** [review]
Looks like we need to use

```python
from onnxscript import optimizer
```

File: torch/onnx/_internal/exporter.py [hunk 1]
```diff
@@ -1277,6 +1277,13 @@ def export(self) -> ONNXProgram:
                     "ONNXScript optimizer is not available. Skipping optimization. "
                     "Please `pip install onnxscript -U` to enable post-export optimization."
                 )
+            except Exception as e:
```
Comment (justinchuby) [inline torch/onnx/_internal/exporter.py:1280]: When there is an exception, onnx_model may have been changed in place already and become invalid.

Comment (justinchuby) [reply torch/onnx/_internal/exporter.py:1280]: Maybe not since it is the output of infer_shape; but I would confirm that to make sure

Comment (thiagocrepaldi) [reply torch/onnx/_internal/exporter.py:1280]: Ok, what is the recommendation? Improve error message to let user know?

Comment (justinchuby) [reply torch/onnx/_internal/exporter.py:1280]: I think the passes make a copy so we should be good. We may need to make the atomic expectation clear. Currently this is not explicit iirc. @gramalingam @BowenBao

Comment (thiagocrepaldi) [reply torch/onnx/_internal/exporter.py:1280]: It is crucial that the optimization is atomic. Either we succeed with a modified graph or we fail and the input model is unchanged. My understanding was that it was atomic already, but if it is not, we really should make it so

Comment (gramalingam) [reply torch/onnx/_internal/exporter.py:1280]: A good point to discuss in the optimizer implementation. It should be safe for the top-level function (`optimizer.optimize`) which does not modify the input model, but returns a new optimized model. It becomes tricky only if we try to eliminate this copy.

Comment (justinchuby) [reply torch/onnx/_internal/exporter.py:1280]: Created https://github.com/microsoft/onnxscript/issues/1476

File: test/onnx/test_fx_to_onnx.py [hunk 2]
```diff
@@ -330,6 +330,7 @@ def forward(self, input):
                 expected_node="aten.mul.Tensor",
             )
 
+    @pytorch_test_common.xfail(error_message="'submodule_1' not found in {")
```
Comment (thiagocrepaldi) [inline test/onnx/test_fx_to_onnx.py:333]: @BowenBao is the value info propagated by the optimize or something like that?

Comment (BowenBao) [reply test/onnx/test_fx_to_onnx.py:333]: What likely happened is the simple local function is optimized away and broke this test assumption.

**Comment by titaiwangms** [comment]
In favor of #127039

**Pull request closed without merging**